tests/unit/test_state.py

```diff
@@ -0,0 +1,178 @@
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[HOSTNAME_CONFIG_NAME] = "example?.com"
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert "consist of alphanumeric and hyphen" in str(err.value)
+
+
+def test_empty_path():
+    """
+    arrange: Mock charm with empty path.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[PATH_CONFIG_NAME] = "   "
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+    assert str(err.value) == "Config error: ['path = : String should have at least 1 character']"
+
+
+def test_invalid_path():
+    """
+    arrange: Mock charm with path with invalid character.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[PATH_CONFIG_NAME] = "/^"
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert (
+        str(err.value)
+        == "Config error: ['path = /^: Value error, Path contains non-allowed character']"
+    )
+
+
+@pytest.mark.parametrize(
+    "invalid_backends, error_message",
+    [
+        pytest.param("", "Empty backends configuration found", id="empty backends"),
+        pytest.param(
+            "mock",
+            "Config error: ['backends = mock: value is not a valid IPv4 or IPv6 address']",
+            id="incorrect backends format",
+        ),
+        pytest.param(
+            "10.10.1",
+            "Config error: ['backends = 10.10.1: value is not a valid IPv4 or IPv6 address']",
+            id="incorrect IP format",
+        ),
+    ],
+)
+def test_config_backends_invalid_backends(invalid_backends: str, error_message: str):
+    """
+    arrange: Mock charm with invalid backends config.
+    act: Create the state from the charm.
+    assert: Configuration error raised with a correct error message.
+    """
+    charm = MockCharmFactory()
+    charm.config[BACKENDS_CONFIG_NAME] = invalid_backends
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert str(err.value) == error_message
+
+
+def test_config_protocol_invalid():
+    """
+    arrange: Mock charm with invalid protocol config.
+    act: Create the state from the charm.
+    assert: Configuration error raised with a correct error message.
+    """
+    charm = MockCharmFactory()
+    charm.config[PROTOCOL_CONFIG_NAME] = "unknown"
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert (
+        str(err.value)
+        == "Config error: [\"protocol = unknown: Input should be 'http' or 'https'\"]"
+    )
+
```

Comment:
I'd test that "http" is also accepted, only "https" has been tested

Reply:
Added a test.